src/notte/common/notifier/mail.py

```diff
@@ -0,0 +1,97 @@
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+from pydantic import BaseModel
+from typing_extensions import override
+
+from notte.common.agent.types import AgentResponse
+from notte.common.notifier.base import BaseNotifier
+
+
+class EmailConfig(BaseModel):
+    """Configuration for email sending functionality."""
+
+    smtp_server: str
+    smtp_port: int = 587
+    sender_email: str
+    sender_password: str
+    receiver_email: str
+    subject: str = "Notte Agent Task Report"
+
+
+class EmailNotifier(BaseNotifier):
+    """Email notification implementation."""
+
+    def __init__(self, config: EmailConfig) -> None:
+        super().__init__()  # Call parent class constructor
+        self.config: EmailConfig = config
+        self._server: smtplib.SMTP | None = None
+
+    async def connect(self) -> None:
+        """Connect to the SMTP server."""
+        if self._server is not None:
+            return
+
+        self._server = smtplib.SMTP(host=self.config.smtp_server, port=self.config.smtp_port)
+        _ = self._server.starttls()
+        _ = self._server.login(user=self.config.sender_email, password=self.config.sender_password)
+
+    async def disconnect(self) -> None:
+        """Disconnect from the SMTP server."""
+        if self._server is not None:
+            _ = self._server.quit()
+            self._server = None
+
+    @override
+    async def send_message(self, text: str) -> None:
+        """Send an email with the given subject and body."""
+        if self._server is None:
+            await self.connect()
+
+        msg = MIMEMultipart()
+        msg["From"] = self.config.sender_email
+        msg["To"] = self.config.receiver_email
+        msg["Subject"] = self.config.subject
+
+        msg.attach(MIMEText(text, "plain"))
+
+        if self._server:
+            _ = self._server.send_message(msg)
+
+    @override
+    async def notify(self, task: str, result: AgentResponse) -> None:
+        """Send an email notification about the task result.
+
+        Args:
+            task: The task description
+            result: The agent's response to be sent
+        """
+        await self.connect()
+        try:
+            body = f"""
+Hello,
+
+This is an automated message from your Notte Agent.
+
+Task Details:
+-------------
+Task: {task}
```

Comment:
*code duplication*. Please tackle this. You should override `def format_message(...) = Hello,
This is an automated message from your Notte Agent. ... super.format_message(...) ... Best regards,
Your Notte Agent 🌒